### PR TITLE
Clean app upgrade

### PR DESCRIPTION
**Pull request by Psycojoker**
Hello,

This is mostly refactoring to make the code more readable. I've simply removed unused code and assign a temp variable for better readability.

We need to do way more refactoring on app.py but that's for later.

### CODE REVIEW

**Comment by Psycojoker** [comment]
You need to stimulate an upgrade with a dummy app, not very convenient.

**Comment by Psycojoker** [comment]
We need another approval.

File: src/yunohost/app.py [hunk 1]
```diff
@@ -349,10 +349,12 @@ def app_upgrade(auth, app=[], url=None, file=None):
...
     if not app:
-        if (not url and not file):
-            app = os.listdir(apps_setting_path)
+        if not url and not file:
+            app = [app["id"] for app in app_list(installed=True)["apps"]]
     elif not isinstance(app, list):
-        app = [ app ]
+        app = [app]
+
+    logger.info("Will upgrade apps %s", ", ".join(app))
```
Comment (opi) [inline src/yunohost/app.py]: nitpicking & personnal taste, but I'd prefer something like "Upgrading apps %s";

**Comment by alexAubin** [comment]
I refactorized the upgradability check to be directly in app_info, in order to simplify the fix of https://dev.yunohost.org/issues/787. Still not really clean but definetely cleaner than current stuff imho :/

**Pull request merged**